src/m365/spo/commands/group/group-member-add.ts

```diff
@@ -212,6 +222,46 @@ class SpoGroupMemberAddCommand extends SpoCommand {
       });
   }
 
+  private async aadUserGet(options: Options, userName: string, logger: Logger, validUserNames: string[]): Promise<CommandOutput> {
+    const aadUserGetCommandoptions: AadUserGetCommandOptions = {
+      ...(options.userName && { userName: userName }),
+      ...(options.email && { email: userName }),
+      output: 'json',
+      debug: options.debug,
+      verbose: options.verbose
+    };
+    const aadUserGetOutput: CommandOutput = await Cli.executeCommandWithOutput(AadUserGetCommand as Command, { options: { ...aadUserGetCommandoptions, _: [] } });
+
+    if (this.debug) {
+      logger.logToStderr(aadUserGetOutput.stderr);
+    }
+
+    validUserNames.push(JSON.parse(aadUserGetOutput.stdout).userPrincipalName);
+
+    return aadUserGetOutput;
```

Comment:
Why do you return the data as well as push it to an array? Seems like too much. I'd say either return the data or push it to an array.